drivers/common/openserial.h

```diff
@@ -59,8 +61,123 @@ enum {
 #define SERFRAME_PC2MOTE_TRIGGERSERIALECHO       ((uint8_t)'S')
 #define SERFRAME_PC2MOTE_COMMAND                 ((uint8_t)'C')
 #define SERFRAME_PC2MOTE_TRIGGERUSERIALBRIDGE    ((uint8_t)'B')
+
+//for the schedule
+#define OPENSERIALMAXNUMCELLS   5
+#define SCHEDULEIEMAXNUMCELLS   3
 
 //=========================== typedef =========================================
+
+enum{
+    SERTYPE_PKT_TX             = 0x01,
+    SERTYPE_PKT_RX             = 0x02,
+    SERTYPE_CELL               = 0x03,
+    SERTYPE_ACK                = 0x04,
+    SERTYPE_PKT_DROPPED        = 0x05,
+    SERTYPE_DIO                = 0x06,
+    SERTYPE_DAO                = 0x07,
+    SERTYPE_NODESTATE          = 0x08,
+    SERTYPE_6PCMD              = 0x09,
+};
+
+enum{
+    CELLADD = 1,
+    CELLDEL = 2
+};
+
+
+BEGIN_PACK
+typedef struct{
+    uint8_t     code;
+    uint8_t     type;
+    uint8_t     shared;
+    uint8_t     slotOffset;
+    uint8_t     channelOffset;
+    uint8_t     neighbor[8];
+} evtCell_t;
+END_PACK
+
+BEGIN_PACK
+typedef struct{
+    uint8_t     code;        // RX or TX
+    uint8_t     l2addr[8];
+}evtAck_t;
+END_PACK
+
+BEGIN_PACK
+typedef struct{
+    uint8_t     length;
+    uint8_t     frame_type;
+    slotOffset_t slotOffset;
+    uint8_t     frequency;
+    uint8_t     l2Dest[8];
+    uint8_t     txPower;
+    uint8_t     numTxAttempts;
+    uint8_t     l3_destinationAdd[16];
+    uint8_t     l3_sourceAdd[16];
```

Comment:
Order of "l3_destinationAdd[16]" and "l3_sourceAdd[16]" needs to be changed, as the order of prints in "LogPktTx()" function.
Also variables for L4Protocol, L4SrcPort and L4DestPort must be defined here.

Reply:
I will handle your comments as soon as I have some time.

Reply:
Sure, I pushed the last version of my C file 👎 

I corrected my mistakes, and tested the current implementation (stats are deactivated by default)